src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -301,12 +311,67 @@ public InnReachResponseDTO recall(String trackingId, String centralCode, RecallD
     return success();
   }
 
+  @Override
+  public InnReachResponseDTO borrowerRenew(String trackingId, String centralCode, BorrowerRenewDTO borrowerRenew) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var hold = transaction.getHold();
+    var loan = requestService.getLoan(hold.getFolioLoanId());
+    var existingDueDate = loan.getDueDate();
+    var requestedDueDate = new Date(borrowerRenew.getDueDateTime() * 1000L);
+
+      try {
+        var renewedLoan = renewLoan(hold);
+        var calculatedDueDate = renewedLoan.getDueDate();
+        if (calculatedDueDate.after(requestedDueDate) || calculatedDueDate.equals(requestedDueDate)) {
+          transaction.setState(BORROWER_RENEW);
+        } else {
+          recallRequestToCentralSever(transaction, existingDueDate);
+        }
+      } catch (Exception e) {
+        if (existingDueDate.before(requestedDueDate)) {
+          recallRequestToCentralSever(transaction, existingDueDate);
+        } else {
+          throw new CirculationException("Failed to renew loan: " + e.getMessage(), e);
+        }
+      }
+
+    return success();
+  }
+
   private InnReachRecallUser getRecallUserForCentralServer(String centralCode) {
     return centralserverRepository.fetchOneByCentralCode(centralCode)
       .map(CentralServer::getInnReachRecallUser)
       .orElseThrow(() -> new EntityNotFoundException("Recall user is not set for central server with code = " + centralCode));
   }
 
+  private void recallRequestToCentralSever(InnReachTransaction transaction, Date existingDueDate) {
+    var trackingId = transaction.getTrackingId();
+    var centralCode = transaction.getCentralServerCode();
+
+    String uri = resolveD2irCircPath(D2IR_ITEM_RECALL_OPERATION, trackingId, centralCode);
+
+    var dueDateForRecallRequest = new HashMap<>();
+    dueDateForRecallRequest.put("dueDateTime", existingDueDate);
```

Comment:
D2IR API handles dates as UNIX epoch **seconds**. Are you sure this will work?

Reply:
added date conversion